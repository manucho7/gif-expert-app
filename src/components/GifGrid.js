import React, { useState, useEffect } from "react";
import GifGridItem from "./GifGridItem";

export const GifGrid = ({ category }) => {

    const [images, setImages] = useState([]);

    useEffect( () => {
        getGifs();
    }, [])

    const getGifs = async () => {

    const url =
        "https://api.giphy.com/v1/gifs/search?q=rick and morty&limit=10&api_key=EKYfdfTGfUEzUTT5g5wl1ONxKqC6u2Rx";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
        return {
            id: img.id,
            title: img.title,
            url: img.images?.downsized_medium.url,
        };
    });

    console.log(gifs);
    setImages( gifs );
    };

//    getGifs();

    return (
        <>
            <h3> { category } </h3>
            <div className="card-grid">


                {
                    images.map( img => (
                        <GifGridItem
                            key={ img.id }
                            { ...img }
                    />
                    ))
                }

            </div>
        </>
    )
};

export default GifGrid;
