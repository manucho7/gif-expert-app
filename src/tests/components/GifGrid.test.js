import '@testing-library/jest-dom';
import '@testing-library/dom';
import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock( '../../hooks/useFetchGifs' );

describe('Pruebas en <GifGrid/>', () => {

    const category = 'One Punch';
    
    test('debe de mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow( <GifGrid category={ category }/> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('debe de mostrar items cuando se cargan images useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost.com',
            title: 'anything'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={ category }/> );

        expect( wrapper ).toMatchSnapshot();

    });
})
