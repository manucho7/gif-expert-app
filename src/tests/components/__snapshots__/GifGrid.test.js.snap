// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid/> debe de mostrar items cuando se cargan images useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="anything"
      url="https://localhost.com"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid/> debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
